src/java.net.http/share/classes/java/net/http/HttpClient.java

```diff
@@ -364,6 +364,16 @@ public static Builder newBuilder() {
          * HTTP client will be bound to an automatically
          * assigned socket address.
          *
+         * <p> Common usages of the {@code HTTPClient} do not require
+         * this method to be called. Setting a local address, through this
+         * method, is only for advanced usages where users of the {@code HTTPClient}
+         * require specific control on which network interface gets used
+         * for the HTTP communication. Callers of this method are expected to
+         * be aware of the networking configurations of the system where the
+         * {@code HTTPClient} will be used and care should be taken to ensure the
+         * correct {@code localAddr} is passed. Failure to do so can result in
+         * requests sent through the {@code HTTPClient} to fail.
+         *
```

Comment:
`{@code HTTPClient}` should probably `{@code HttpClient}` since that's how the class is named.

Reply:
You are right, that was an oversight. I've now fixed it in the latest update to this PR.